### PR TITLE
Setting Renderers, FlowPanel Improvements, and Setting/About Views

### PR DESCRIPTION
**Pull request by dlamkins**
Introduces proper setting renderer views which can automate the display and wiring of supported setting types.  Currently those include:

- Bool
- String
- KeyBinding
- Enum (any)

Improvements have been made to the Container and FlowPanel controls to allow for these new renderers.

Containers can now specify a sizing method for both width and height.  This allows them to auto-size based on their child contents, fill the remaining space, or the normal behavior.

FlowPanel has additional flow rules allowing for "TopToBottom" now as well as single width column flows and single height row flows.

These features have been used to replace the static builders that were previously being used for the Overlay Settings panel and About panel.  These have now been replaced with actual views.

### CODE REVIEW

File: Blish HUD/Strings/GameServices/OverlayService.Designer.cs [hunk 1]
```diff
@@ -144,7 +156,7 @@ internal class OverlayService {
         }
         
         /// <summary>
-        ///   Looks up a localized string similar to Minimize to tray when Guild Wars 2 Closes.
+        ///   Looks up a localized string similar to Minimize to tray when Guild Wars 2 Closes..
```
Comment (greaka) [inline Blish HUD/Strings/GameServices/OverlayService.Designer.cs:159]: Is this intended?

File: Blish HUD/GameServices/Settings/UI/Views/SettingsView.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                _lockBounds = value;
+
+                UpdateBoundsLocking(_lockBounds);
+            }
+        }
+
+        private ViewContainer _lastSettingContainer;
+
+        public SettingsView(SettingCollection settings, int definedWidth = -1)
+            : base(new SettingEntry<SettingCollection>() { Value = settings }, definedWidth) {
```
Comment (greaka) [inline Blish HUD/GameServices/Settings/UI/Views/SettingsView.cs]: Can't you use `: this(...` here and make it a noop?

**Review by greaka — APPROVED** [review]

File: Blish HUD/Strings/GameServices/OverlayService.Designer.cs (hunk 1 above)
Comment (dlamkins) [reply Blish HUD/Strings/GameServices/OverlayService.Designer.cs:159]: It definitely looks odd, but it's autogenerated.  The actual string is "Minimize to tray when Guild Wars 2 Closes." which I did purposefully add a period to.  Looks like the autogenerated comment just uses `Looks up a localized string similar to {0}.`

File: Blish HUD/GameServices/Settings/UI/Views/SettingsView.cs (hunk 2 above)
Comment (dlamkins) [reply Blish HUD/GameServices/Settings/UI/Views/SettingsView.cs]: Good thinking.  Fixed!

File: Blish HUD/Strings/GameServices/OverlayService.Designer.cs (hunk 1 above)
Comment (greaka) [reply Blish HUD/Strings/GameServices/OverlayService.Designer.cs:159]: Alright, just wanted to make sure

**Review by greaka — APPROVED** [review]

**Pull request merged**